content/en/blog/2023/waypoint-proxy-made-simple/index.md

```diff
@@ -0,0 +1,221 @@
+LISTENER              CHAIN                                                 MATCH                                         DESTINATION
+envoy://connect_originate                                                       ALL                                           Cluster: connect_originate
+envoy://main_internal inbound-vip|9080||reviews.default.svc.cluster.local-http  ip=10.96.104.108 -> port=9080                 Inline Route: /*
+envoy://main_internal direct-tcp                                            ip=10.244.2.14 -> ANY                         Cluster: encap
+envoy://main_internal direct-tcp                                            ip=10.244.1.6 -> ANY                          Cluster: encap
+envoy://main_internal direct-tcp                                            ip=10.244.2.11 -> ANY                         Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.2.11 -> application-protocol='h2c'  Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.2.11 -> application-protocol='http/1.1' Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.2.14 -> application-protocol='http/1.1' Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.2.14 -> application-protocol='h2c'  Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.1.6 -> application-protocol='h2c'   Cluster: encap
+envoy://main_internal direct-http                                           ip=10.244.1.6 -> application-protocol='http/1.1'  Cluster: encap
+envoy://connect_terminate default                                               ALL                                           Inline Route:
+{{< /text >}}
+
+For requests arriving on port `15008`, which by default is Istio’s inbound {{< gloss >}}HBONE{{< /gloss >}} port, the waypoint proxy terminates the HBONE connection and forwards the request to the `main_internal` listener to enforce any workload policies such as AuthorizationPolicy. If you are not familiar with [internal listeners](https://www.envoyproxy.io/docs/envoy/latest/configuration/other_features/internal_listener), they are Envoy listeners that accepts user space connections without using the system network API. The `--waypoint` flag added to the `istioctl proxy-config` command, above, instructs it to show the details of the `main_internal` listener, its filter chains, chain matches, and destinations.
+
+Note `10.96.104.108` is the reviews' service VIP and `10.244.x.x` are the reviews v1/v2/v3 pod IPs, which you can view for your cluster using the `kubectl get svc,pod -o wide` command. For plain text or HBONE terminated inbound traffic, it will be matched on the service VIP and port 9080 for reviews or by pod IP address and application protocol (either `ANY`, `h2c`, or `http/1.1`).
```

Comment:
```suggestion
Note `10.96.104.108` is the reviews' service VIP and `10.244.x.x` are the reviews' v1/v2/v3 pod IPs, which you can view for your cluster using the `kubectl get svc,pod -o wide` command. For plain text or HBONE terminated inbound traffic, it will be matched on the service VIP and port 9080 for reviews or by pod IP address and application protocol (either `ANY`, `h2c`, or `http/1.1`).
```